The Globe and Mail.js

```diff
@@ -1,223 +1,159 @@
-			var day = metaTags["pubdate"].substr(6, 2);
-			var year = metaTags["pubdate"].substr(0, 4);
-			
-			newItem.date = (year + "-" + month + "-"+ day);
-		}
-	
-	//tags	
-		if (metaTags["article_keywords"]) {
-			tagsContent = metaTags["article_keywords"].split("; ");
-		}
-		
-		for (var i = 0; i < tagsContent.length; i++) {
-		     	if (tagsContent[i] != (" ") && tagsContent[i] != ("")) {
-			     	newItem.tags[i] = tagsContent[i];
-		     	}
-	     	}	
-		
-		associateMeta (newItem, metaTags, "headline", "title");
-		associateMeta (newItem, metaTags, "summary", "abstractNote");
-		associateMeta (newItem, metaTags, "desk", "section");	
-		associateMeta (newItem, metaTags, "article_id", "callNumber");
-		associateMeta (newItem, metaTags, "credit", "rights");
-	
-	//rest of author (shared between both newspaperArticle types)
-		if (author) {
-			
-		
-			if (author.substr(0,3).toLowerCase() == "by ") {
-				author= author.substr(3);
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (!items) {
+				return true;
 			}
-		
-			var authors = author.toLowerCase().split(" and ");
-			for (var i=0; i<authors.length; i++) {
-				var author = authors[i];
-				var words = author.split(" ");
-				
-				for (var j=0; j<words.length; j++) {
-					if (words[j] != "") {
-						words[j] = words[j][0].toUpperCase() + words[j].substr(1).toLowerCase();
-					}
-				}
-				author = words.join(" ");
-				newItem.creators.push(Zotero.Utilities.cleanAuthor(author, "author"));			
+			var articles = [];
+			for (var i in items) {
+				articles.push(i);
 			}
-		}
-	}	
+			ZU.processDocuments(articles, scrape);
+		});
+	} else {
+		scrape(doc, url);
+	}
+}
+
+function scrape(doc, url) {
 	
-	newItem.url = doc.location.href;
-	newItem.complete();
+	newItem = new Zotero.Item("newspaperArticle");
+	newItem.url = url;
+
+	//get headline
+	var title = ZU.xpathText(doc, '//article/h1');
+	if (!title) title = ZU.xpathText(doc, '//meta[@property="og:title"]/@content');
+	newItem.title = title;
+
+	//get abstract
+	newItem.abstractNote = ZU.xpathText(doc, '//meta[@property="og:description"]/@content');
 	
-}
+	//get date
+	var xpathdate_author = '//time[@itemprop="datePublished"]';
+	var date = ZU.xpathText(doc, xpathdate_author);
+	if (date) {
+		newItem.date = ZU.strToISO(date);
+	}
 
-function doWeb(doc, url) {
-	var namespace = doc.documentElement.namespaceURI;
-	var nsResolver = namespace ? function(prefix) {
-		if (prefix == 'x') return namespace; else return null;
-	} : null;
+	//get author or organization
+	//get author or organization
+	var authors = ZU.xpath(doc, '//meta[@itemprop="author"]/@content');
+	for (var i in authors){
+		newItem.creators.push(ZU.cleanAuthor(authors[i].textContent, "author"));
+		newItem.creators[i].publisher = ZU.xpathText(doc, '//article/header//div[@itemprop="publisher"]//p').replace("\n",'');
```

Comment:
I used publisher in the loop because for every author there is a publisher tag, in case there are multiple authors of an article, they may have different entries in publishers. For some articles, the publisher tag is Reuters where the author contains content as "Wire Service". Also, there are articles where the publisher tag has content "Special to The Globe and Mail". How to handle such cases?
I have updated the code as you suggested but the newItem.publisher is not visible in the output of the test cases( a '-' sign appears before it in Scaffold)